simple_network_sim/sampleUseOfModel.py

```diff
@@ -8,25 +8,30 @@
 from pathlib import Path
 from typing import Optional
 
+from data_pipeline_api.api import API, DataAccess, ParameterRead
+from data_pipeline_api.file_system_data_access import FileSystemDataAccess
+
 from . import network_of_populations as ss, loaders
 
 # Default logger, used if module not called as __main__
 logger = logging.getLogger(__name__)
 
 
 def main(argv):
+    api = API(FileSystemDataAccess("data_pipeline_inputs", "metadata.toml"))
+
     t0 = time.time()
 
     args = build_args(argv)
     setup_logger(args)
     logger.info("Parameters\n%s", "\n".join(f"\t{key}={value}" for key, value in args._get_kwargs()))
 
     network = ss.createNetworkOfPopulation(
-        diseasesProgressionFn=args.compartment_transition,
-        populationFn=args.population,
-        graphFn=args.commutes,
-        ageInfectionMatrixFn=args.mixing_matrix,
-        movementMultipliersFn=args.movement_multipliers,
+        api.read_table("human/compartment-transition", version=1),
```

Comment:
My initial thoughts about the createNetworkOfPopulation would be that you'd pass "human/compartment-transition" to it, rather than the table. The idea being that the user of the model should be restricted to always use the data api rather than providing their own data. To sort of gurantee that there are no "sneaky" ad-hoc data being used by mistake.

Reply:
But what about inputting custom parameters when doing sensitivity analysis or inference?

Reply:
In principle alternative parameterisations would be managed via a config file that the data pipeline API will read, but the calls in the code will remain unchanged.

Reply:
What do you mean by "different parametrizations"? Do you mean different model parameters, or different models used altogether?

Currently, my use cases, which we'll need for all the parameter sensitivity and inference project, is to run the simulation with e.g. a different compartment transition matrices - so same parametrization, but different parameters, if that makes sense.
Right now to do this I need to create the default model, then manually create my parameters (in the notebook) and manually recreate a network, not using `createNetworkOfPopulation`. This seems exactly opposite of the Rafael is mentioning `the model should be restricted to always use the data api rather than providing their own data.`
So we need to think about a safe way to do so?

I think storing all the parameter samples is not economical/practical, we should be able to generate them on the fly and discard them. What do you think?

Reply:
I meant different parameter values. The data pipeline API is really just a particular way of loading data from files, so it shouldn't intrude too much. I can't quite tell what you mean, but are you saying that you want to load hyperparameters of distribution over networks, and then sample the network in the model? I don't think the pipeline itself has anything to say about that - that is purely a modelling choice, we just have to find a place/name for the relevant parameters.

Reply:
I agree with you that maybe it's not a pipeline question, maybe more a question for Rafael, I'll chat with him separately

Reply:
I think I'll just take the Dataframes as parameters of createNetworkOfPopulations. My idea was that it'd be nice for the constructor for the object to just take the parameter names from the perspective of guaranteeing only data from the data api is loaded. But I guess @FiodorG has the requirement of loading arbitrary data, also, it makes things easier to test.